plugins/node/instrumentation-kafkajs/src/instrumentation.ts

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  SpanKind,
+  Span,
+  SpanStatusCode,
+  Context,
+  propagation,
+  Link,
+  trace,
+  context,
+  diag,
+  ROOT_CONTEXT,
+} from '@opentelemetry/api';
+import {
+  MESSAGINGDESTINATIONKINDVALUES_TOPIC,
+  MESSAGINGOPERATIONVALUES_PROCESS,
+  MESSAGINGOPERATIONVALUES_RECEIVE,
+  SEMATTRS_MESSAGING_SYSTEM,
+  SEMATTRS_MESSAGING_DESTINATION_KIND,
+  SEMATTRS_MESSAGING_DESTINATION,
+  SEMATTRS_MESSAGING_OPERATION,
+} from '@opentelemetry/semantic-conventions';
+import type * as kafkaJs from 'kafkajs';
+import type {
+  EachBatchHandler,
+  EachMessageHandler,
+  Producer,
+  RecordMetadata,
+  Message,
+  ConsumerRunConfig,
+  KafkaMessage,
+  Consumer,
+} from 'kafkajs';
+import { KafkaJsInstrumentationConfig } from './types';
+import { VERSION } from './version';
+import { bufferTextMapGetter } from './propagator';
+import {
+  InstrumentationBase,
+  InstrumentationModuleDefinition,
+  InstrumentationNodeModuleDefinition,
+  safeExecuteInTheMiddle,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+
+export class KafkaJsInstrumentation extends InstrumentationBase<
```

Comment:
Is it possible to hide the kafkajs types from public API?
see https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/GUIDELINES.md

Reply:
Removed the internal type from exports